### PR TITLE
Add Issue Template

### PR DESCRIPTION
**Pull request by FireController1847**
It should help new users out with being able to report issue and give us some organization.

I'd love to add in the comments of the template links to guides on how to get the output log and savegame, if there are any.

You can see an example of this working if you head over to [FireController1847/Cities-Skylines-Traffic-Manager-President-Edition#issues](https://github.com/FireController1847/Cities-Skylines-Traffic-Manager-President-Edition/issues) and create a new issue.

Also, I'd be up to make a template for feature requests too, if that floats your boat. We can make other templates for other things as well :)

### CODE REVIEW

**Comment by aubergine10** [comment]
Yes, this is a great idea!!!

Reviewing....

**Comment by aubergine10** [comment]
@FireController1847 : Although it looks really nice once the issue is submitted, I think currently the text in the new issue form is too much, especially with all that markdown syntax.

Maybe something simpler like:

```
## Describe problem

Type here

## Link to savegame

Paste URL here

<!-- How to share savegame? See: https://bit.ly/2FXog0z -->

## Link to log file

Paste URL here 

<!-- How to share log file? See: https://bit.ly/2I3YVFL -->

## Sceenshots

Optional: Paste URLs here

<!-- How to share screenshots? See: https://bit.ly/2Kc8owO -->

## Additional information

Provide any other information / links here
```

While it might result in a bit of visible boilerplate landing in the issue, I think it will be much easier for users to work with, What do you think?

**Comment by aubergine10** [comment]
Also, could you add one for feature requests? Maybe something like:

```
## Describe your idea:

Type here

## Any links with additional information?

Paste URLs here

## Sceenshots illustrating how it works?

Optional: Paste URLs here

<!-- How to share screenshots? See: https://bit.ly/2Kc8owO -->

## Additional information:

Provide any other information / links here
```

**Comment by aubergine10** [comment]
And one for translations?

```
Looking for file format information? See: https://bit.ly/2FVUglO

## Which language?

Type here

## Localisation file

If you're familiar with GitHub you can send us a Pull Request. (recommended)

Or, click on a *blank line* in this text box and then drag-and-drop the files to directly upload them.

Or, paste a URL to the file here

## Additional information:

Provide any other information / links here
```

Note: I have no clue how the localisation stuff works - I'll need help with the following wiki page: https://bit.ly/2FVUglO

**Comment by FireController1847** [comment]
When I get back home I'll take a look at making some changes.

**Comment by aubergine10** [comment]
I found info on the localisation stuff so updated that wiki page myself; if someone could check over it:

https://github.com/krzychu124/Cities-Skylines-Traffic-Manager-President-Edition/wiki/Localisation

**Comment by aubergine10** [comment]
I think there's still too much markdown, and also having the basic instructional text in html comments is confusing. I think its OK for the subsequent "how to" text (eg, see my examples above) as there the comment sort of separates it out, making it look more like a quote or note from user perspective.

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+name: Bug Report
+about: Help us understand what we need to fix.
+title: ''
+labels: ''
+assignees: ''
+---
+
+<!-- Before submitting this issue, please search and make sure it hasn't already been reported, thanks! -->
```
Comment (aubergine10) [inline .github/ISSUE_TEMPLATE/bug-report.md]: GitHub will suggest possible existing issues as they type in their details, so I don't think we need this.

Comment (FireController1847) [reply .github/ISSUE_TEMPLATE/bug-report.md]: Okay

**Comment by FireController1847** [comment]
@aubergine10 Question: In particular is it the example? The comments below each question make sense, imho, even if they're somewhat in markdown. However, I can see how the example could confuse people.

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+---
+name: Feature Request
+about: Make suggestions on how TM:PE could improve.
+title: ''
+labels: 'feature'
```
Comment (aubergine10) [inline .github/ISSUE_TEMPLATE/feature-request.md]: add `triage` label too?

(the triage label means that we can quickly list any issues that none of the team have looked at; after looking at an issue, we can remove the triage label)

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+---
+
+<!-- Before submitting this feature, please search and make sure it hasn't already been suggested, thanks! -->
+
+### Description
+<!-- In a few sentences, summarize what your idea is. -->
+
+
+### Sources / Links
+<!-- Post any links below that describe how your issue works. For example, with turn on red it'd be the link to a "turn on red rule" in a driver's handbook; things like that. -->
```
Comment (aubergine10) [inline .github/ISSUE_TEMPLATE/feature-request.md]: Good idea with the example! ❤️

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+### Screenshots
+<!-- If you have screenshots showing how this feature works, put them below. How to: https://bit.ly/2Kc8owO -->
+
+
+### Additional Info
+<!-- Any other things worthy of being noted? Let us know what they are below. -->
+
+
+<!--
+EXAMPLE:
```
Comment (aubergine10) [inline .github/ISSUE_TEMPLATE/feature-request.md]: Again, not sure an example is helpful; it just adds visual clutter to the 'new issue' form.

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+<!-- In a few sentences, summarize what your idea is. ->
+Right on red (or left on red if map in left hand drive mode) should allow vehicles to turn at a red light after stopping.
+
+### Sources / Links
+<!-- Post any links below that describe how your issue works. For example, with turn on red it'd be the link to a "turn on red rule" in a driver's handbook; things like that. ->
+https://dld.utah.gov/wp-content/uploads/sites/17/2018/12/Driver-Handbook-2018-2019.pdf#page=29
+
+### Screenshots
+<!-- If you have screenshots showing how this feature works, put them below. How to: https://bit.ly/2Kc8owO ->
+![image](https://user-images.githubusercontent.com/9030085/44266955-c67dc180-a26f-11e8-94e4-35e18dc05e3d.png)
```
Comment (aubergine10) [inline .github/ISSUE_TEMPLATE/feature-request.md]: we're getting in to realms of teaching people github while they are trying to make a feature request

suggest having a read of: https://alistapart.com/article/flywheelsandfriction :)

File: .github/ISSUE_TEMPLATE/translations.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+---
+name: Translations
+about: Help us make TM:PE available for everyone, no matter what language.
+title: ''
+labels: 'localisation'
```
Comment (aubergine10) [inline .github/ISSUE_TEMPLATE/translations.md]: also add a `triage` label?

File: .github/ISSUE_TEMPLATE/translations.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+### Localization File
+<!-- If you're familiar with github, go ahead and send us a Pull Request instead of submitting this issue. If not, click below this comment then drag-and-drop your translation file to upload it here. Alternatively, you can paste the URL to the file below. -->
+
+
+### Additional Information
+<!-- Want to let us know anything else? Tell us below. -->
+
+
+<!--
+EXAMPLE:
```
Comment (aubergine10) [inline .github/ISSUE_TEMPLATE/translations.md]: Again, example is probably overkill

**Comment by FireController1847** [comment]
Okay, I think I've gotchya. Give me a couple minutes and I'll make these template issues great again.

**Comment by FireController1847** [comment]
@aubergine10 Okay, so I cut down on the comments massively. How this?

**Comment by aubergine10** [comment]
@FireController1847 did you push your additional changes? I see lots of "Pending" flags on the comments above but no subsequent changes?

**Comment by FireController1847** [comment]
@aubergine10 I committed a bunch after my initial comment "I think I've gotchya," implementing said changes.

**Comment by aubergine10** [comment]
In the early days of the internet, search engines looked like this:

yahoo

And everyone was generally confused because they were presented with all kinds of stuff that was superfluous to their requirement to just search the internet.

Then google came along and said "people just want the search box" and made a search engine that looks like this:

google
 
And everyone started using it because it didn't throw the [kitchen sink](https://dictionary.cambridge.org/dictionary/english/everything-but-except-the-kitchen-sink) at them.

Then Microsoft came along with Bing, which is arguably a better search engine than google, but they made it look like this:

bing

So everyone still uses google because it doesn't throw the kitchen sink at them.

The same happened with mobile phones. They used to have clunky interfaces that showed too much, then Apple came along and simplified it with a touch screen. Everyone who could afford it ditched their Nokias and similar devices and moved over to Apple because Apple was easier to use. Google clocked on pronto, and made their OS super clean too ([material](https://material.io/)) - in fact, cleaner than Apple in most cases, and cheaper too. So now most people are moving over to Android because it's cleaner, easier and cheaper.

All of these transitions have something in common: Minimalist Gestalt

Instead of more is more, minimalist gestalt teaches that less is more.

Put another way:

> When everything has focus, nothing does

This is what I've been trying to convey in the reviews of this PR.

There are actually hundreds of gestalt principles, but most people focus on [7 main principles](https://www.usertesting.com/blog/gestalt-principles/).

Gestalt is just one piece of the mix, it focuses on the visual psychology. It's also important to reduce friction in general, which is why I posted the link to [Flywheels, kinetic energy, and friction](https://alistapart.com/article/flywheelsandfriction/), an article which caused a seismic shift (or more acutely, a [gestalt shift](https://www.roangelo.net/logwitt/gestalt-shift.html)) in UX thinking when it was published.

These things apply to all software and even hardware - anything that has to interact with humans.

If something needs a link to an example showing how to do it, it is designed wrong. If a user wants to perform a task, like report a bug, that has to be the absolute primary focus of the interface.

If we start asking them to do a bunch of stuff they don't want to do, or start putting hurdles in their way, they will leave and not report the bug - they don't want the kitchen sink thrown at them.

Make sense?

**Comment by FireController1847** [comment]
@aubergine10 Yes, and I understand what you are talking about (I've taken a Digital Media Design college class). However, in this situation it has been simplified greatly. Every item is to the point, and does not contain unnecessary instruction. In fact, the only thing that still could be considered "too much" here would be the "replace me" things. However, those are placeholders, NOT instructions.

Some people learn best by example -- not to say that they need to _learn_ how to do it, but most people aren't smart enough to figure a lot of this out. No matter how simple it is, I would still provide an example on how to do it, because that's how I've always understood to do things. I'm not saying everyone has to read about the example; let alone go to the URL. You know that won't happen. But I want the option to be there for people to know how to do it.

Not to say I won't disagree it's a little much in terms of the instructions (placeholders), people are just _not smart enough_ to be able to figure out what to put there. That's why I provide the instruction for (almost) every comment; people WILL do it wrong if we don't tell them how to do it, no matter what. In this situation, I'd assume you're meaning it's a little complex for the user's to figure out why there's need for the "replace me" instructions. I don't know how much more simplistic I can make it without outright removing them (and when I do that, people are not smart and often don't put it in the right spot). "R E P L A C E M E" is direct instruction on exactly what to do. This is not UX design, nor is this UI design. This is a textbox that people are typing in. They need to know what we're asking for and where to put it. That's what I have provided with these templates.

The whole template is extremely simplistic; the required ones simply state what to do, and we need to tell the user that the other ones are OPTIONAL (they often can't make that inference on their own). I think we're both going for the same goal, but with different outlooks. You're focusing on "bad UX design," and I'm focusing on "there's only so much 'design' we can put in text."

Additionally, people like to know why they need to go through the work of listing their mods or putting in things like their output log. That's the main reason I have the explanation: is for if users need to understand WHY they should go through the effort. Without those, people will likely skip important information needed to figure out what the bug is. Listing mods is an important aspect when reporting a bug. If they can't list the mods they had enabled, why should we immediately assume it was TM:PE's fault? Same with me asking them to disable all mods and see if it happens with only TM:PE. I'm eliminating things we'd have to repeat and say over and over after they've made the issues and instead putting an, extremely simplistic at this point mind you, one-line sentence explaining three things: why they need it, how to do it, and where to put it. That's it. Each sentence is explaining those things, in one line.

Google's search engine is not text-based. And, it's a poor example. "Search" is a one-word term that actually DOES show up within the search box. They have a UI that can symbolize where to move the cursor to type, and they can use colors, icons, and shadows to guide the user where to go. We don't have that luxury here. We have to base it entirely off text. So, all the instruction that _would_ get guided by UI needs to be converted to text. This is as simplistic as I can make it.

Sorry if I repeated myself. I just don't understand how what I have now is _still_ too complex for a **text-based user-input box without any UI or UX** and **requires a lot of information to ensure we get a proper bug instead of a waste of time and space.**

**Comment by FireController1847** [comment]
@krzchu124 We've added three in this PR; one for. Bug Reports, one for Feature Requests, and one for Translation Submissions. What other categories would you suggest?

**Comment by FireController1847** [comment]
@krzychu124 My apologies, I had forgotten to merge my branch into master so you could see it working. Now if you go to my branch and create a new issue, you should see it in action.

**Comment by FireController1847** [comment]
@aubergine10 @krzychu124 Alright, fine. I've taken your approach. Do you think this will work?

**Comment by aubergine10** [comment]
> Simplify Extremist

Will check in a moment, just sorting another PR

**Comment by aubergine10** [comment]
I've updated the links in the wiki so they now use the new issue templates :)

**Comment by krzychu124** [comment]
How nice that automatically GitHub set up everything for us 😄

**Pull request merged**